src/graphql/schemas/args/argGenerator.ts

```diff
@@ -0,0 +1,177 @@
+import { ClassType, Field, InputType } from "type-graphql";
+import { SortOrder } from "../enums/sortEnums.js";
+import {
+  BigIntSearchOptions,
+  BooleanSearchOptions,
+  IdSearchOptions,
+  NumberArraySearchOptions,
+  NumberSearchOptions,
+  StringArraySearchOptions,
+  StringSearchOptions,
+} from "../inputs/searchOptions.js";
+
+type ReferenceDefinition = {
+  type: keyof SearchOptionType;
+  references: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    entity: ClassType<any>;
+    fields?: Record<string, keyof SearchOptionType>;
+  };
+};
+
+export type SearchOptionType = {
+  string: typeof StringSearchOptions;
+  number: typeof NumberSearchOptions;
+  bigint: typeof BigIntSearchOptions;
+  id: typeof IdSearchOptions;
+  boolean: typeof BooleanSearchOptions;
+  stringArray: typeof StringArraySearchOptions;
+  numberArray: typeof NumberArraySearchOptions;
+};
+
+export const SearchOptionMap = {
+  string: StringSearchOptions,
+  number: NumberSearchOptions,
+  bigint: BigIntSearchOptions,
+  id: IdSearchOptions,
+  boolean: BooleanSearchOptions,
+  stringArray: StringArraySearchOptions,
+  numberArray: NumberArraySearchOptions,
+} as const;
+
+// TODO: a type cache is needed to avoid creating the same types multiple times
+// I mean, do we have to?
+// Cache for storing generated types
+export const typeCache: Record<
+  string,
+  ReturnType<typeof createEntityArgs>
```

Comment:
Any reason to type out `ReturnType<typeof createEntityArgs>` instead of just `EntityArgs`?

Reply:
This is currently only used in test files for the mock entity generates to ensure type safe type as a function of a method.